File: backend/schemas/task.py
from pydantic import BaseModel
from typing import Optional, Any

class TaskCreate(BaseModel):
    name: str
    workflow_id: int

class TaskStatus(BaseModel):
    status: str

    class Config:
        orm_mode = True
class TaskResponse(BaseModel):
    id: int
    name: str
    status: str
    result: Optional[Any] = None
    workflow_id: int

    class Config:
        orm_mode = True

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'in-progress' | 'completed' | 'failed';

interface BaseTaskConfig {
  id: number;
  name: string;
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
}

export interface AIImageGenerationConfig extends BaseTaskConfig {
  type: 'ai-image-generation';
  prompt: string;
  width: number;
  height: number;
}

export interface ApiTaskConfig extends BaseTaskConfig {
  type: 'api';
  endpoint: string;
  payload?: Record<string, unknown>;
}

export interface EmailTaskConfig extends BaseTaskConfig {
  type: 'email';
  to: string;
  subject: string;
  body: string;
}

export type TaskConfig = AIImageGenerationConfig | ApiTaskConfig | EmailTaskConfig;

export interface TaskCreate {
  workflowId: number;
  name: string;
  type: 'ai-image-generation' | 'api' | 'email';
  config?: TaskConfig;
}

export interface TaskResponse extends TaskCreate {
  id: number;
  status: TaskStatus;
  result?: Record<string, unknown>;
}
